core/src/main/java/org/mule/api/context/notification/FlowStackElement.java

```diff
@@ -13,17 +13,17 @@
  * 
  * @since 3.8.0
  */
-public interface FlowStackElement extends Serializable, Cloneable
+public interface FlowStackElement extends Serializable
 {
 
     /**
-     * Deep-copies this instance.
+     * @return the path of the currently executing processor in the current flow.
      */
-    FlowStackElement clone();
+    String currentMessageProcessor();
```

Comment:
Not sure this method name is correct... "current" only applies to the top element of the stack.